docu: Rework connector kit adoption view (#1300)
eclipse-tractusx/eclipse-tractusx.github.io: docs-kits/kits/connector-kit/adoption-view/adoption-view.md

### Mission

Collaboration needs trust. Within Catena-X trust is built on open standards and technologies that transparently provide features to find data, to agree on the usage of the data in a legally binding way and to provide technical means for accessing the data.

The Connector KIT provides all necessary information and technology references to perform trustful data exchange within Catena-X. It bundles the interaction patterns, relevant standards, APIs, and reference implementations for developers.

## Important terms and definitions

Before going into details, some basic terms need to be described as they are used throughout the KIT.

- Dataspace: The overall term for the principles with which data transfer is organized within Catena-X. The technical details of those principles are the scope of this KIT. It is based on concepts developed by the [IDSA][idsa-url].
- Data Provider: A data provider in Catena-X is a company, that owns data and offers that data to others.
- Data Consumer: A data consumer in Catena-X is a company, that consumes data offered by a data provider.
- Dataspace participant: A company acting as data provider or data consumer or both.

> - Connector: Is the common term for a technical component that is formally named (dataspace) participant agent, see [CX-0018][Catena-X-Standards-url] for reference. It provides central features needed to act as data provider or consumer within the Catena-X dataspace.

Reviewer: did you want to link to CX-0018 directly here? https://catenax-ev.github.io/docs/standards/CX-0018-DataspaceConnectivity

Author: Actually, issue is, that there is no stable link to a concrete version, that is why I currently only refer to the standards page, as it is easy to find the standard from there.

---
sidebar_position: 1
title: Adoption View
description: The adoption view describes the basic infrastructure of cross-enterprise data exchange
id: adoption-view
---

![Connector kit banner](@site/static/img/kits/connector/connector-kit-logo.svg)

> The Connector KIT describes the basic infrastructure that is used for any cross-enterprise data exchange within Catena-X. The infrastructure is based on the [Dataspaces][idsa-url] and [Self-Sovereign Identities][self-sovereign-identity-url] as basic mechanisms to achieve the targeted goals.

Author: done

## Vision & Mission

### Vision

Sharing data along the automotive supply chain adds great value for every participant. A wide range of use cases support efficiency in engineering, production and logistics. A prerequisite of the support of the transfer use cases is trust between the involved partners and a clear scope on which data is transfered and under which conditions it can be used. The technologies described in the Connector KIT provide exactly that framework that allows trustworthy exchange within the automotive supply chain.

### Mission

Data transfer between companies needs trust. But what exactly does trust mean. From a data provider side, it requires that data is only given to authenticated consumers under defined conditions for the usage of the data. From a consumer side, the data used has to be trustworthy, i.e., the source of the data is trusted and the semantics of the data transfered is defined. For both sides, it is important to allow and easy integration into existing IT landscapes and the usage of a wide variety of technologies to meet the diverse transfer use cases.

> Although there are established solutions in the internet, the chosen dataspace technology provides the required features scalable and with a quality that exceeds the established solutions. Main advantages are:

Author: Reworked the section to get the point clearer

> - Scalability through the usage of Self-Sovereign Identities. This reduces efforts in the management as the complex n:m relationship between providers and consumers is reduced to the management of ones own identity that is provided by a central issuance agency. The concept favorably supports the concept of business identities, as the data transfer happens between companies and not between single users.

Reviewer 2: you'd have to obtain a separate identity from each of your business partners - usually in the form of users in an IdP.

> - The conditions under which the data is allowed to be used are transparently stated during the authentication process, they can be complete in themselves or refer to existing contracts, but are associated directly to a data transfer authorization. This allows flexibility and transparency in the definition of different transfer models depending on the involved data and partners.

Reviewer 3: I would at least add a third main reason why data spaces are favourable over existing solutions on the internet - especially over platforms - and that's the decentral nature of a data space. There's no single point of failure and no central entity that controls and oversees transfers. This holds true for the orchestration of transfers but also for the transfer itself: Data isn't stored centrally but data providers and consumers decide for themselves where they want to store the data.

> - The conditions under which the data is allowed to be used are transparently stated during the authentication process, they can be complete in themselves or refer to existing contracts, but are associated directly to a data transfer authorization. This allows flexibility and transparency in the definition of different transfer models depending on the involved data and partners.

Author: done

The mission of the Connector KIT is to provide all necessary information and technology references to perform trustful data exchange within Catena-X. It bundles the interaction patterns, relevant standards, APIs, and reference implementations for developers. The concrete scope within Catena-X is shown in the following figure by the green boxes.

![Scope of the KIT](./assets/kit_scope.drawio.png)

## Important terms and definitions

Before going into details, some basic terms need to be described as they are used throughout the KIT.

- Dataspace: The overall term for the principles with which data transfer is organized within Catena-X. The technical details of those principles are the scope of this KIT. It is based on concepts developed by the [IDSA][idsa-url].
- Data Provider: A data provider in Catena-X is a company, that owns data and offers that data to others.
- Data Consumer: A data consumer in Catena-X is a company, that consumes data offered by a data provider.
- Dataspace participant: A company acting as data provider or data consumer or both.

> - Connector: Is the common term for a technical component that is formally named (dataspace) participant agent, see [CX-0018][Catena-X-Standards-url] for reference. It provides central features needed to act as data provider or consumer within the Catena-X dataspace.

Reviewer 3: We are spreading out the introduction of terms and concepts across the various sections. Here we introduce provider/consumer... and below under "Basic values and principles" we introduce DSP and policies. We don't introduce offers or contracts. Does it make sense to once introduce the most common terms (such as here https://eclipse-dataspace-protocol-base.github.io/DataspaceProtocol/2025-1-RC4/#terminology) and then work with them?

Author: I did this actually on purpose, the terms are specified when they are needed. I do not like the gathering in one section because it disturbs the flow of reading and encourages to accept half-understood terms. And the places, which define terms represent actually the three levels of abstraction.
1. General concepts on a bird's eye view
2. Main technical concepts on a level below
3. Concrete technical aspects of the chosen technologies

That is how I would prefer to be lead through the introduction.

Reviewer 3: With that bit of explanation that makes senso to me. Proposal: Shall we have a very short chapter that introduces the "structure of the document"? Such as in a scientific paper? I.e. "The following document introduces the concepts of xxx by first zzz then zzz. As this is a complex and for many readers new topic, terms and definitions are only introduced when they are needed to not overwhelm the reader. 

Something along that lines?

## Basic values and principles

> The data exchange in Catena-X follows a set of values and derived principles that drive the interaction patterns. In detail, they are:

Reviewer 3: Observation: Those principles and values have a larger overlap with the advantages over existing technologies ;)

Author: I have moved the advantages from the mission to better places lower in the document

  - which type of data is offered and what to expect semantically as content of the data,
  - how the data is transfered, i.e., which protocols resp. which APIs to use for accessing it,
  - under which conditions the data can be used.

## Basic Concepts used for Dataspace Connectivity

The basic concept followed in Catana-X to achieve the targeted goals is the use of dataspace technology which is based on the existence of a connector. A connector provides the technical implementation of the dataspace principles using standardized protocols and assets maintained by the [Eclipse Dataspace Working Group][edwg-url]. The assets are based on concepts created by the [IDSA][idsa-url] and are in accordance to principles of [Gaia-X][gaiax-url].

In detail, the relevant assets are:

- [Dataspace Protocol (DSP)][dsp-url]: The DSP is a specified protocol for the interaction of a data consumer with a data provider in order to find data offerings, get access to the data and managing the data transfer. In Catena-X, the protocol is used with its HTTP binding, i.e., the requests and responses defined in the DSP are translated to REST API endpoints.

> - Identity: The DSP requires trustworthy identities for dataspace participants. For Catena-X, there are bascially two relevant identifier which represent identity. A *Business Partner Number* for a legal entity (BPNL) represents an unique identifier for a company in a legally binding fashion. For technical reasons, a second identifier is needed based on the [*Decentralized Identifier* (DID)][did-url] standard. This identifier type, especially the used WEB-DID method allows to connect metadata like public keys or central service references to the company by providing the information in a DID document. This is hosted by a participant and the endpoint address for retrieval of the document is derived from the (WEB-)DID only.

Reviewer 3: Can you explain in one sentence for a non SSI-Expert why we need both: DIDs and BPNLs? Would make that section more readable!

> - Identity: The DSP requires trustworthy identities for dataspace participants. For Catena-X, there are bascially two relevant identifier which represent identity. A *Business Partner Number* for a legal entity (BPNL) represents an unique identifier for a company in a legally binding fashion. For technical reasons, a second identifier is needed based on the [*Decentralized Identifier* (DID)][did-url] standard. This identifier type, especially the used WEB-DID method allows to connect metadata like public keys or central service references to the company by providing the information in a DID document. This is hosted by a participant and the endpoint address for retrieval of the document is derived from the (WEB-)DID only.

Author: Hope I covered all

- Verifiable Credentials: The identity and the acceptance of general terms and conditions for the participation in Catena-X are expressed in the form of a verifiable credential. A verifiable credential is basically a digitally signed data structure which allows a participant to proof the availability of a certain fact like the participants identity or the acceptance of terms and conditions in a tamper-resistant way. Verifiable credentials are typically issued by an authority which is authorized to issue the defined credential and, therefore, is assumed as trustworthy.

> - [Self-Sovereign Identity][self-sovereign-identity-url]: The self-sovereign identity mechanism used is based on the principle of self-management of the own identity in the form of *Verifiable Credential* (VC) in an own wallet. As a consequence, in authentication scenarios, no third party is involved. The consumer can simply provide his credentials which can be proven by the provider with simply knowning the public key of the issuing authority as shown in the following picture.

Author: I cannot follow that one, I introduce SSI just after introducing VCs and earlier DIDs. What is your point here, it brings together the concepts into a mechanism for authentication, I have to explain it, and I do not see how I can do that earlier.

> - [Self-Sovereign Identity][self-sovereign-identity-url]: The self-sovereign identity mechanism used is based on the principle of self-management of the own identity in the form of *Verifiable Credential* (VC) in an own wallet. As a consequence, in authentication scenarios, no third party is involved. The consumer can simply provide his credentials which can be proven by the provider with simply knowning the public key of the issuing authority as shown in the following picture.

Author: I have reworked it, hopefully it is improved

- Discovery: Based on the DID of a data provider, Catena-X defines a decentral discovery mechanism that allows to identify endpoints of connectors in order to initiate data transfer use cases. The discovery mechanism uses the DID document to provide information on connector endpoints again in a decentral manner.

The described technologies are complete to initiate and conduct data transfers. The prerequisite needed to get started are the knowledge of the identifiers of the data provider, both the BPNL and the DID. As a consequence, Catena-X defines additional mechanisms that allow to retrieve these informations. These mechanisms are out-of-scope for the Connector KIT. We refer to the [business partner KIT][BPDM-url] for further informations.

### Reasoning of the usage of the dataspace concept

> There are established interaction patterns in the internet, which have already proven their applicability in many different use cases of typical interactions. An important technology in this regard is OAuth2 which is the de-facto standard for authentication and authorization. It needs good arguments to motivate, why for the data transfer scenarios targeted at within Catena-X a new technology is needed.

Reviewer 3: I see why you are proposing this chapter - but who is the audience here? Most "adopters" (we are in the adoption view here") don't question the design choices of Catena-X but they want to understand how it works. The adoption view should provide a high level introduction into how the CX Dataspace works and not loose itself in justifications why oAuth2 isn't enough for a&a.

> The rationale behind the need for using dataspace technology is basically driven from the underlying principles described above. Major advantages of the dataspace technologies are:

Reviewer 3: We already have a chapter on "main advantages" at the very top of this chapter. Should/can we merge them together?

- It does not require any technical relationship prior to the execution of the data transfer scenario. The only information needed by a data consumer are the identifiers of the data provider which can be retrieved by central dataspace services. From there on, everything is within the scope of the dataspace technology. Especially, there is no need to exchange metadata like the endpoints to access the data or the semantics of data access.

>   Although dataspaces do not get rid of some central discovery mechanisms, alternative solutions either require substantially more centralization, e.g., a central identity provider that contradicts the decentralization principle, or they require an amount of upfront peer-to-peer information exchange, that limits the scalability to a non-acceptable level as each data transfer partner needs to be managed and, e.g., secrets like passwords need to be exchanged on a regular basis. The dataspace technology on the contrary solves those issues out-of-the-box without a lot of effort needed on data provider as well as consumer side.

Reviewer 3: Consistency: One chapter above you write about "Discovery: ... decentral discovery mechanism" and now it says that we don't get rid of central discovery mechanisms". As a first time reader that would confuse me. However, I don't have a proposal for a better approach. Something like "mostly decentral"?

- The applied pattern for authentication and authorization is applicable to a whole range of different technologies, standardzing the pure protocol for gaining access to a data offer. E.g., the technology supports standard REST API access control using tokens. In this case the end result, a token does not really divert from the way OAuth2 handles the access to a resource server. But the connector can also handle other ways of interaction, like push scenarios, where data is pushed from the provider to the consumer in a one time fashion or as a stream of data. The technical means, like a bucket to store a pushed object or a technology to receive a constant stream of information has to be provided, but the overall mechanism to get access and to manage the data transfer is standardized within the mechanisms of the Dataspace Protocol.

In general, the dataspace technology supports in a favorable way the principles described above as general targets of Catena-X.

> - Dencentralization: As mentioned, as soon as the identifierts of the data provider are known, the whole process does not require any interaction with a central service. The few remaining accesses to central services are specified in a way, that bascially no conrete interest in certain information can be deducted from those requests.

Reviewer 2: ```suggestion
- Decentralization: As mentioned, as soon as the identifierts of the data provider are known, the whole process does not require any interaction with a central service. The few remaining accesses to central services are specified in a way, that bascially no conrete interest in certain information can be deducted from those requests.
```

> - Dencentralization: As mentioned, as soon as the identifierts of the data provider are known, the whole process does not require any interaction with a central service. The few remaining accesses to central services are specified in a way, that bascially no conrete interest in certain information can be deducted from those requests.

Reviewer 3: This again seems to be a repetition of what has already been said at the "Priciples" section. At leas there's a larger overhead. I would propose merging those sections.

     Push transfers are by default a one time transfer, i.e., a transfer process is initiated, the provider pushes the contracted data to the sink and finalizes the transfer process. An additional transfer mechanism to be established soon is an infinite push, i.e., the transfer process is not terminated and the provider sends new data elements as soon as they are created until the process is stopped manually or by expiry of the contract.

After the transfer process is started, the data plane is responsible for the data transfer, i.e., it mediates between the consuming service and the providing service to get the data transfered. The data plane is a transfer technology specific implementation that connects an existing technology to the concepts of the dataspace concept. The technology specific information on data endpoints used and access information needed are part of the transfer process creation.

### Important Terms

> #### Contract Definition

Reviewer 2: This starts to blur the lines between terminology from EDC and DSP. Perhaps it'd make sense to divide the `Important Temrs` section?

Author: Right @arnoweiss, but I think, that at this point, the DSP abstractions are a bit too abstract and not the way people talk about the elements in a dataspace. I have no idea, how we can support readers with those abstractions, that is why I used the ones used here.

@gerbigf I don't see this as unstructured, but as something with a logical buildup. As I wrote, I typically don't feel supported by a general terminology section because it requires me to jump around in the document.

#### Transfer Process

After a successful contract agreement, a *Transfer Process* can be started from the consumer connector to initiate a data transfer based on a contract agreement. A transfer process is represented by a state machine as well. It is asynchronous by nature, i.e., once started it can be used multiple times to execute a transfer of data from provider to consumer. It can be terminated manually or suspended and resumed at any time. Some transfer types like one-time push terminate the process automatically. Transfer processes can be started as long as a contract assigment has not expired.

The transfer process carries access information between consumer and provider. Depending on the direction of the data transfer, push or pull, access credentials, like tokens need to be sent, so that the active part can transfer the data from the data source to the data sink.

### Interaction Overview

The following figure shows the involved services and the basic interaction between the services.

> ![Basic Service Interaction](./assets/connector_interactions.drawio.png)

Author: I have to admit, I am not getting your request on IAM. It is am implementation detail, imho, what technology is used in a vcertain setup. I do not see the IDP you mention in your picture as a relevant component.

> ![Basic Service Interaction](./assets/connector_interactions.drawio.png)

Reviewer 4: Fair point

> ![Basic Service Interaction](./assets/connector_interactions.drawio.png)

Reviewer 4: Now we can read it super!

The figure shows the interaction between a consumer *Participant A* and a provider *Participant B*. In the following, the different interactions are described in detail.

The first interaction is done on the provider side, by preparing the data and creating the contract definition to offer the data to potential consumers.

- P1: Manage the offered data offers and policies. Build contract definitions by combining offered data with an usage and an access policy.

> - P2: Provide the data according to the offered contract definitions, e.g., operate a resource server like a *Digital Twin Registry* and fill it with the mapping data between digital twins and the connected sub models.

Reviewer 3: Is the DT registry really a good example here? External readers could think that the registry not only holds references but actual data? 

```suggestion
- P2: Provide the data according to the offered contract definitions, e.g., operate a resource server such as a REST API that provides CO2 values of certain parts.
```

With these steps, data is available and offered in a way that it can be found by consumers and access can be negotiated. The *Provider Business Application* is a service that allows to manage provided data, e.g., manually or for larger setups, by an automation that, driven by data created in an internal IT system, executes all necessary registrations and data transformations to prepare the new data for retrieval by a consumer.

> The data transfer is initiated at the consumer. If the consumer does not have the necessary details, to initiate the transfer, he has to check some centrally operated services (step R1), that provide him with the necessary identifiers (BPNL and DID) to initiate the decentral data transfer. The central services work in a way, that it is impossible to track the concrete interest in a company, e.g., the mentioned *BDRS* (BPN-DID Resolution Service) provides a list of all known mappings between BPNLs and DIDs. It is a local mechanism in the consumer connector to identify the concrete mapping needed in a certain situation.

Reviewer 3: "concrete interest in a company" or "of a company"? If it's the first, that might need additional information such as "it's not possible to derive which information a consumer is seeking, only that he wants to learn about the endpoints of a provider?

Based on the identifiers, the data transfer can be negotiated. This is done with the following steps:

> - N1: The *Consumer Business Application* requests a step as defined in the Dataspace Protocol to be executed by the *Control Plane* of the consumer connector.

Reviewer 3: "requests a step as defined..." sounds a bit ominous - can we be more precise?

- N2: The *Control Plane* downloads the DID document typically stored in the *Identity Wallet* of the provider. It uses the information in the DID to identify the URL under which the document is accessible. In the current Catena-X dataspace, the DID document is cached in the central *Portal*. This is a historical legacy to be overcome in one of the ecosystem releases.
  The whole step can be omited if the DID document is already cached.

> - N3: To initiate a DSP request, the consumer *Control Plane* has to provide an access token, that allows the providers *Control Plane* to retrieve verifiable credentials in order to check whether the request is valid and all prerequisites are met. This token is provided by the *Identity Wallet* of the consumer.

Reviewer 5: ```suggestion
- N3: To initiate a DSP request, the consumer *Control Plane* has to provide an access token, that allows the providers *Control Plane* to retrieve verifiable credentials in order to check whether the request is valid and all prerequisites are met. This token is provided by the consumer's *Wallet*.
```

- N4: The consumer *Control Plane* executes the intended DSP request by callig the corresponding DSP endpoint at the provider *Control Plane*. With the request, the consumer sends the token retrieved in step N3.
- N5: The provider *Control Plane* retrieves the verifiable credentials necessary to validate the legitimacy of the request by calling the *Identity Wallet* of the consumer using the token created in step N3. The control plane gets access to the credentials and checks their validity. In addition, the control plane checks other conditions and if successful executes the request and provides an appropriate response to the consumer control plane.

> The steps N1 to N5 are bascially the same for every DSP call, be it to request the catalog, to initiate a contract negotiation or to start of a transfer request. The Consumer Business Application executes the required DSP requests in the appropriate order basing the next request on data received in the previous request. In the day-to-day business situations, the dominating action on DSP level is the management of transfer requests, as the catalog retrieval and the contract negotiation are only necessary, if access to a new data offering is required. If the concept is applied efficiently, such requests are the exception, as most data can be retrieved using existing contracts.

Reviewer 3: ```suggestion
The steps N1 to N5 are bascially the same for every DSP call, be it to request the catalog, to initiate a contract negotiation or to start a transfer request. The Consumer Business Application executes the required DSP requests in the appropriate order basing the next request on data received in the previous request. In the day-to-day business situations, the dominating action on DSP level is the management of transfer requests, as the catalog retrieval and the contract negotiation are only necessary, if access to a new data offering is required. If the concept is applied efficiently, such requests are the exception, as most data can be retrieved using existing contracts.
```
Not sure myself - is it "management of transfer requests" or "transfer processes"?

#### Reference Implementation

Eclipse Tractus-X provides a reference implementation of a connector in the repository [tractusx-edc][tractusx-edc-url]. The reference implementation provides a full fledged implementation of the described features and principles, ready to be used in productive environments. The implementation is based on the [Eclipse Dataspace Components connector][edc-url] implementation, which is by itself a reference implementation for the specifications provided by the [EDWG][edwg-url].

The reference implementation offers a non-standardized API for a dataspace participant to manage the operated connector. More information on how the reference implementation can be used in your environment are given in the [development-view][dev-view-url].

> ### Contract Definition Design

Author: Absolutely, that will be part of the extra page. I named is asset-design, but actually, the name is not that important, it is short and easy to handle, but for an enduser it will not be visible.

An important aspect for an efficient usage of the dataspace technology is the design of contract definitions. Due to the longevity of contract agreements and the overhead implied by catalog requests and contract negotiations, it makes sense to create data offers which provide not only one data point. In general, it should not be necessary to negotiate for the same type of data more than once.

The aspect of contract definition design is discussed in an [extra page][asset-url] of the adotion view.

### Discovery of Connectors

> Catena-X provides a decentral discovery mechanism based on the DID and the possibility to download a DID document that contains, besides other things, references to typed service endpoints. The DID document is managed by the wallet and allows to add the necessary metadata by an API. The document is a digitally signed data structure that can be retrieved from anyone without authentication needs.

Reviewer 5: ```suggestion
Catena-X provides a decentral discovery mechanism where connector endpoints are contained in a participant's DID document.
A DID document is public and can be accessed by anyone who has knowledge of the corresponding DID.
```

DID documents are not signed. There's not really a need for it since I can proof control over a DID document through knowledge of my private key.

- [CX-0018 Eclipse Data Space Connector (EDC)][Catena-X-Standards-url]
- [CX-0049 DID Document Schema][Catena-X-Standards-url]
- [CX-0050 Catenea-X Specific Credential][Catena-X-Standards-url]
- [CX-0152 Policy Constraints For Data Exchange][Catena-X-Standards-url]

## Notice

This work is licensed under the [CC-BY-4.0](https://creativecommons.org/licenses/by/4.0/legalcode).

- SPDX-License-Identifier: CC-BY-4.0
- SPDX-FileCopyrightText: 2024 Contributors of the Eclipse Foundation
- SPDX-FileCopyrightText: 2025 Cofinity-X GmbH
- Source URL: [https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/blob/main/docs-kits/kits/connector-kit/adoption-view/adoption-view.md](https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/blob/main/docs-kits/kits/connector-kit/adoption-view/adoption-view.md)
- Images: [https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/blob/main/docs-kits/kits/connector-kit/adoption-view/assets](https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/blob/main/docs-kits/kits/connector-kit/adoption-view/assets)

[edc-url]: https://eclipse-edc.github.io

[edwg-url]: https://dataspace.eclipse.org

> [did-url]: https://www.w3.org/TR/did-1.0

Reviewer 5: ```suggestion
[did-url]: https://www.w3.org/TR/did-1.0

[did-web-url]: https://w3c-ccg.github.io/did-method-web/
```

See comment L228.

reliable framework that ensures the principles targeted at by Catena-X. That is the reason why all use cases defined
by Catena-X are required to use the connector technology in order to facilitate the data exchange scenarios of the
use case.

## Business Architecture

The dataspace protocol suggests a *Connector*, which implements the state machines and interactions patterns
specified in the protocol. A connector is thereby separated into two distinguished concepts the *Control Plane*,
which implements the protocol interactions, and a *Data Plane*, which realizes the data transfer with a specific
transfer protocol. The control plane as such is a static concept which can be implemented and used in multiple
transfer scenarios, whereas the data plane is a virtual concept that is provided in multiple facets in order to
implement different data transfer types, like standard http rest api access, transferring binary large objects, or
continuous data streaming.

The following sequence diagram shows the general interaction patterns of the control plane, explained afterwards in
more detail.

```mermaid

> sequenceDiagram
>     Consumer Control Plane->>+Provider Control Plane: Catalog Request
>     Provider Control Plane->>-Consumer Control Plane: Catalog
> 
>     Consumer Control Plane->>Provider Control Plane: Contract Negotiation Request
>     Consumer Control Plane->>+Provider Control Plane: Contract Negotiation Status?
>     Provider Control Plane->>-Consumer Control Plane: Status of Contract Negotiation Process
> 
>     Consumer Control Plane->>+Provider Control Plane: Start Transfer Process
> 
>     alt Pull Scenario
>         Consumer Control Plane->>+Provider Control Plane: Status Transfer Process?
>         Provider Control Plane->>-Consumer Control Plane: Status of Transfer Process + EDR if started
>     else Push Scenario
>         Provider Control Plane->>Provider Data Plane: Initiate data Transfer
>     end

Reviewer 4: ```suggestion
sequenceDiagram
  participant CBA as Consumer Business Application
  participant CCP as Consumer Control Plane
  participant PCP as Provider Control Plane
  participant PDP as Provider Data Plane

  CCP->>+PCP: Catalog Request
  PCP->>-CCP: Catalog

  CCP->>PCP: Contract Negotiation Request
  CCP->>+PCP: Contract Negotiation Status?
  PCP->>-CCP: Status of Contract Negotiation Process

  CCP->>+PCP: Start Transfer Process

  alt Pull Scenario
      CCP->>+PCP: Status Transfer Process?
      PCP->>-CCP: Status of Transfer Process + EDR if started
      CBA->>CCP: Wait for Negotiation <br>Status==FINALIZED then get EDR
      CBA->>CCP: Get Authorization Token
      CBA->>PDP: Initiate data Transfer
  else Push Scenario
      PCP->>PDP: Initiate data Transfer
  end
  alt Data Exchange
      PDP->>CBA: Response proxied from asset endpoint
  end
```

@lgblaumeiser In this way we can see the difference between pull and push. But I am open to add it into another section or diagram.

<img width="946" height="780" alt="image" src="https://github.com/user-attachments/assets/bc233d7e-3c16-4dc5-ac35-91012a1948c2" />

I have also resumed for you the participants so we can use variables making easier to maintain.

Reviewer 4: Understood then we can leave the interactions of the CBA with the CCP out. 

I have an idea

transfer scenarios, whereas the data plane is a virtual concept that is provided in multiple facets in order to
implement different data transfer types, like standard http rest api access, transferring binary large objects, or
continuous data streaming.

The following sequence diagram shows the general interaction patterns of the control plane, explained afterwards in
more detail.

```mermaid
sequenceDiagram
    Consumer Control Plane->>+Provider Control Plane: Catalog Request
    Provider Control Plane->>-Consumer Control Plane: Catalog

    Consumer Control Plane->>Provider Control Plane: Contract Negotiation Request
    Consumer Control Plane->>+Provider Control Plane: Contract Negotiation Status?
    Provider Control Plane->>-Consumer Control Plane: Status of Contract Negotiation Process

    Consumer Control Plane->>+Provider Control Plane: Start Transfer Process

    alt Pull Scenario
        Consumer Control Plane->>+Provider Control Plane: Status Transfer Process?
        Provider Control Plane->>-Consumer Control Plane: Status of Transfer Process + EDR if started
    else Push Scenario
        Provider Control Plane->>Provider Data Plane: Initiate data Transfer

>     end

Author: Actually, it is described below, but from a sequence diagram perspective, this is it, more details can be added to the development view, imho, but not here.
